spanner/transaction.go

```diff
@@ -1105,19 +1126,15 @@ func NewReadWriteStmtBasedTransaction(ctx context.Context, c *Client) (*ReadWrit
 // Commit tries to commit a readwrite transaction to Cloud Spanner. It also
 // returns the commit timestamp for the transactions.
 func (t *ReadWriteStmtBasedTransaction) Commit(ctx context.Context) (time.Time, error) {
```

Comment:
Would you also need a `CommitWithOptions`? Especially given the nature of #3056.

Reply:
No, because we can pass `CommitOptions` to a transaction via `TransactionOptions` (calling `NewReadWriteStmtBasedTransactionWithOptions`). 

Later, for this `Commit` method, we can have two options: 

1. Update `Commit` to return `(CommitResponse, error)` (a breaking change)
2. Add a new method `CommitReturnResponse` to return `(CommitResponse, error)` (not a breaking change)

Reply:
Ah ok, thanks for explaining.

As for the `Commit`, it would be neat if we could have #1, but since this is a public method, to avoid the risk of inconveniencing users, I think we'll have to go with #2. We could call it `CommitWithStats` in line with `QueryWithStats` but the worry is that we might return something other than stats in the future.

Reply:
Yes, `CommitWithStats` is very specific. We need a very generic name here.